src/trunk.c

```diff
@@ -5377,7 +5377,11 @@ trunk_compact_bundle(void *arg, void *scratch_buf)
       }
       trunk_log_node_if_enabled(&stream, spl, &node);
 
-      should_continue = trunk_compact_bundle_node_has_split(spl, req, &node);
+      should_continue = trunk_compact_bundle_node_has_split(spl, req, &node)
+                        && trunk_key_compare(spl,
+                                             trunk_max_key(spl, &node),
+                                             POSITIVE_INFINITY_KEY)
```

Comment:
So with your above proposal, this should be the max key for the compaction, right?